source/core/index.ts

```diff
@@ -1330,6 +1333,17 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 		delete options.request;
 		delete options.timeout;
 
+		const requestOptions = options as any; // Temporary workaround
+
+		// If `ipVersion` is not present do not override `family`
```

Comment:
But... you know it defaults to `0`? So the last `else if` is unnecessary.

Reply:
I was thinking on a way to keep `family` is the user used that property.

But then I added it to the defaults...
If `ipVersion` has no default from Got `family` is keeped, if I remove the last `else if` it will be overwritten.

Reply:
> I was thinking on a way to keep family is the user used that property.

Then you need to change the last `else if`:

```
} else if (options.ipFamily === 'auto' && !('family' in options)) {
```

Reply:
The logic I am trying to have is:
`family = ipVersion !== undefined ? translate(ipVersion) : family`

Where `translate` is:
- `auto` => 0
- `ipv4` => 4
- `ipv6` => 6